lib/entities/comment.ts

```diff
@@ -24,16 +26,47 @@ export type CommentSysProps = Pick<
   parentEntity: Link<'Entry'> | VersionedLink<'Workflow'>
 }
 
+export enum CommentNode {
+  Document = 'document',
+  Paragraph = 'paragraph',
+  Mention = 'mention',
+}
+
+export interface Mention {
+  nodeType: CommentNode.Mention
+  data: { target: Link<'User'> }
+}
+
+export interface RootParagraph extends Node {
+  nodeType: CommentNode.Paragraph
+  content: (Text | Mention)[]
+}
+
+export interface RichTextDocument extends Node {
+  nodeType: CommentNode.Document
+  content: RootParagraph[]
+}
+
 export type CommentProps = {
   sys: CommentSysProps
   body: string
 }
 
+export type RichTextBodyParam = { body: RichTextDocument }
+
+export type RichTextCommentProps = Omit<CommentProps, 'body'> & RichTextBodyParam
+
+export type CommentBodyFormatHeader = Record<'x-contentful-comment-body-format', 'rich-text'>
```

Comment:
[change request]
I guess we want to allow passing both values explicitly even though `plain-text` is also the default case.
```suggestion
export type CommentBodyFormatHeader = Record<'x-contentful-comment-body-format', 'rich-text' | 'plain-text'>
```

Reply:
I see that right now you need it this way for the overloaded function in `common-types`. I'd suggest adding another overload and using the types `CommentPlainTextBodyHeaders` and `CommentRichTextBodyHeaders`